src/db/query-builder.ts

```diff
@@ -38,8 +67,9 @@ export class QueryBuilder<T extends Schema> {
 		operation: WhereNode<T>["operation"],
 		value: T[keyof T]
 	): QueryBuilder<T> {
-		this.queryObject.where?.push({ key, operation, value });
-		return this;
+		const currQuery = this.clone();
+		currQuery.queryObject.where?.push({ key, operation, value });
+		return currQuery;
```

Comment:
The entire object doesn't necessarily need to be cloned here, just the where field.

```suggestion
                const newQuery = new QueryBuilder(this.queryObject);
                newQuery.where = [...this.queryObject.where, {key, operation, value}];
                return newQuery;
```

Something like this would probably save a bit of memory usage.